src/components/Datepicker/Datepicker.tsx

```diff
@@ -0,0 +1,215 @@
+import React, {useState, useEffect} from "react";
+//@ts-ignore
+import {default as ReactDatePicker, registerLocale} from "react-datepicker";
+import enGB from "date-fns/locale/en-GB";
+import StyledDatePickerWrapper from "./DatePicker.styles";
+import Popover from "../Popover";
+import Button from "../Button";
+import Box from "../Box";
+import ClickAwayListener from "../ClickAwayListener";
+import Typography from "../Typography";
+import {customStyles} from "../../core/styleFunctions";
+import {Modifier} from "react-popper";
+import {Placement} from "@popperjs/core";
+
+export interface DatePickerProps {
+  defaultDate?: Date | null;
+  defaultTime?: string;
+  /**
+   * Callback fired when the date changes
+   */
+  onChange?: (date: Date | null) => void;
+  onTimeChange?: (time: string, date: Date | null) => void;
+  onClose?: () => void;
+  popoverCs?: customStyles;
+  isBlock?: boolean;
```

Comment:
Suggestion; `isBlock` isn't really clear as a name. Would suggest a more descriptive name and/or supporting comments to make it clear what the purpose is.